From f6870a8d7297d562a3015b39525c05ab7b1f3325 Mon Sep 17 00:00:00 2001
From: Rander Wang <rander.wang@linux.intel.com>
Date: Wed, 3 Jul 2019 10:45:13 +0800
Subject: [PATCH] FROMGIT: Revert "ASoC: SOF: hda: set autosuspend delay for hda bus
 device"

This reverts commit 69964d0992a29b95b63e519f2dd7ded2e953fd88

This patch causes a bug: playback and catpure don't work
if playback and catpure are started when ubuntu is just set up.
The hda codec is suspended immediately after initialized in
legacy hda driver and then it works after resume. This patch
breaks the flow in SOF, so just remove it.

Signed-off-by: Rander Wang <rander.wang@linux.intel.com>
---
 sound/soc/sof/intel/hda-codec.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/sound/soc/sof/intel/hda-codec.c b/sound/soc/sof/intel/hda-codec.c
index edac308fcf9e..c711792534da 100644
--- a/sound/soc/sof/intel/hda-codec.c
+++ b/sound/soc/sof/intel/hda-codec.c
@@ -138,7 +138,6 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address)
 int hda_codec_probe_bus(struct snd_sof_dev *sdev)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
-	struct hda_bus *hbus = sof_to_hbus(sdev);
 	int i, ret;
 
 	/* probe codecs in avail slots */
@@ -155,9 +154,6 @@ int hda_codec_probe_bus(struct snd_sof_dev *sdev)
 		}
 	}
 
-	/* set autosuspend delay for hda bus device */
-	snd_hda_set_power_save(hbus, SND_SOF_SUSPEND_DELAY_MS);
-
 	return 0;
 }
 EXPORT_SYMBOL(hda_codec_probe_bus);
-- 
2.17.1

